### ISSUE TITLE
az webapp config source zipdeploy ,doesn't honor  AZURE_CLI_DISABLE_CONNECTION_VERIFICATION to ignore certs for Kudu Zip deploy

### ISSUE THREAD

**Issue by sanchitmehta** [Azure/azure-cli#8770]
**Describe the bug**
When deploying to Azure App Service via zip deploy. Az CLI doesn't honor the environment variable AZURE_CLI_DISABLE_CONNECTION_VERIFICATION=1 to disable the SSL verification and still checks for certs.

Here is the stack trace for the same: 

c:\build>az webapp deployment source config-zip --resource-group "<some resource group>" --name "<app name>" --src "<package name>.zip" --slot "staging"
Connection verification disabled by environment variable AZURE_CLI_DISABLE_CONNECTION_VERIFICATION
C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\site-packages\urllib3\connectionpool.py:847: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\site-packages\urllib3\connectionpool.py:847: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
Connection verification disabled by environment variable AZURE_CLI_DISABLE_CONNECTION_VERIFICATION
C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\site-packages\urllib3\connectionpool.py:847: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\site-packages\urllib3\connectionpool.py:847: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
Connection verification disabled by environment variable AZURE_CLI_DISABLE_CONNECTION_VERIFICATION
C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\site-packages\urllib3\connectionpool.py:847: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\site-packages\urllib3\connectionpool.py:847: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
HTTPSConnectionPool(host='<app name>.scm.<host>', port=443): Max retries exceeded with url: /api/zipdeploy?isAsync=true (Caused by SSLError(CertificateError("hostname '<app name>.scm.<host>' doesn't match either of '*.<host>', '<host>'",),))

**Expected behavior**
Az CLI should completely ignore the certs when the environment variable AZURE_CLI_DISABLE_CONNECTION_VERIFICATION is set.

**Comment by alvinsjtu** [comment]
Hell Team, any update for this issue?

**Comment by willbrim** [comment]
Hi team,
Do we have any update on this or a projected timeline?

**Comment by panchagnula** [comment]
Not yet, added the tentative timeline for this release.

**Comment by willbrim** [comment]
Thank you for the update!

**Comment by yugangw-msft** [comment]
#9125 fixes it

**Comment by savkris** [comment]
Hi all, 

I am facing the same issue as above.

Running az cli behind a corporate proxy 
- set proxy and  set 'AZURE_CLI_DISABLE_CONNECTION_VERIFICATION' to ignore verification
I am able to login and query some commands so far, however "az webapp config source zipdeploy" does not honor this setting.

Above I can see that this issue has been merged into DEV branch. After downloading and installing the dev branch edge az cli build, the issue persist. 

Log below: 
`The command failed with an unexpected error. Here is the traceback:

HTTPSConnectionPool(host='cloudstoragemanager.scm.asemelshared.extnp.national.com.au', port=443): Max retries
y SSLError(SSLError("bad handshake: SysCallError(10054, 'WSAECONNRESET')",),))
Traceback (most recent call last):
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-6e8pr1ug\urllib3\urllib3\contrib\pyopenssl.py", line
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-1smlbq03\pyOpenSSL\OpenSSL\SSL.py", line 1915, in do
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-1smlbq03\pyOpenSSL\OpenSSL\SSL.py", line 1639, in _r
OpenSSL.SSL.SysCallError: (10054, 'WSAECONNRESET')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-6e8pr1ug\urllib3\urllib3\connectionpool.py", line 59
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-6e8pr1ug\urllib3\urllib3\connectionpool.py", line 80
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-6e8pr1ug\urllib3\urllib3\connection.py", line 344, i
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-6e8pr1ug\urllib3\urllib3\util\ssl_.py", line 347, in
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-6e8pr1ug\urllib3\urllib3\contrib\pyopenssl.py", line
ssl.SSLError: ("bad handshake: SysCallError(10054, 'WSAECONNRESET')",)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-1smlbq03\requests\requests\adapters.py", line 449, i
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-6e8pr1ug\urllib3\urllib3\connectionpool.py", line 63
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-6e8pr1ug\urllib3\urllib3\util\retry.py", line 399, i
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='cloudstoragemanager.scm.asemelshared.extnp.nation
i/zipdeploy?isAsync=true (Caused by SSLError(SSLError("bad handshake: SysCallError(10054, 'WSAECONNRESET')",)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-1smlbq03\knack\knack\cli.py", line 206, in invoke
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-1smlbq03\azure-cli-core\azure\cli\core\commands\__in
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-1smlbq03\azure-cli-core\azure\cli\core\commands\__in
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-1smlbq03\azure-cli-core\azure\cli\core\commands\__in
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-1smlbq03\six\six.py", line 693, in reraise
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-1smlbq03\azure-cli-core\azure\cli\core\commands\__in
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-1smlbq03\azure-cli-core\azure\cli\core\commands\__in
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-1smlbq03\azure-cli-core\azure\cli\core\__init__.py",
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-1smlbq03\azure-cli-appservice\azure\cli\command_modu
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-1smlbq03\requests\requests\api.py", line 116, in pos
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-1smlbq03\requests\requests\api.py", line 60, in requ
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-1smlbq03\requests\requests\sessions.py", line 533, i
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-1smlbq03\requests\requests\sessions.py", line 646, i
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-1smlbq03\requests\requests\adapters.py", line 514, i
requests.exceptions.SSLError: HTTPSConnectionPool(host='cloudstoragemanager.scm.asemelshared.extnp.national.c
pdeploy?isAsync=true (Caused by SSLError(SSLError("bad handshake: SysCallError(10054, 'WSAECONNRESET')",),))`

Do I have the incorrect build, do we know if this is fixed?

Thank you in advance

**Comment by panchagnula** [comment]
The change is already released. How are you setting the AZURE_CLI_DISABLE_CONNECTION_VERIFICATION - this is an environment variable, so before you run the command make sure the environment variable is set - if this is being set via command line remember you need to restart the command line terminal or start a new session for the environment variable is set - if the variable is set correctly CLI commands do a display a message saying that, this  environment variable is set.  from the above log it doesn't seem to have any environment variable set. Please check the environment is set before running the command. Thank you.

**Comment by savkris** [comment]
thank you for the prompt response

I do set the variables before starting the session. Same error remains. 
steps:

- set HTTPS_PROXY 
- set AZURE_CLI_DISABLE_CONNECTION_VERIFICATION (to any value, so I set it to :true)
- open cmd: az login (works), az account list (works) , 
- az deploy webapp (fails with the log above)

Any ideas?

**Comment by savkris** [comment]
Do I need to create a new issue to get some answers?

**Comment by yugangw-msft** [comment]
@savkris, 2 questions
1. Are you at the latest CLI
2. If yes, do other commands work?

**Comment by savkris** [comment]
@yugangw-msft 
1. yes I am, tried a dev build just incase, then went back to latest version
2. other commands work

**Comment by yugangw-msft** [comment]
@savkris, please open a new issue. Also want to confirm, w/o by-passing the cert check, the same  command works right?

**Comment by savkris** [comment]
@yugangw-msft 
- if not setting AZURE_CLI_DISABLE_CONNECTION_VERIFICATION I am not able to use any commands. See error below
`Please ensure you have network connection. Error detail: HTTPSConnectionPool(host='login.microsoftonline.com', port=443): Max retries exceeded with url: /742dfa5f-eb9c-4ee1-8d89-cb13b714528c/oauth2/token (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')],)",),))`

- when setting then other commands work (az login, az account list, webapp deployment list..) but not webapp zip deploy

**Comment by yugangw-msft** [comment]
Okay, so 2 things
1. @savkris, your proxy server uses a certificate not fully signed, so you have to by-pass it 
2. @panchagnula, please forward to Kudu team for why the endpoint `cloudstoragemanager.scm.asemelshared.extnp.nationi` decides to close the connections prematurely.

**Comment by savkris** [comment]
thanks for the quick reply @yugangw-msft 
I tried the workaround with the certificate by manually adding it to the cacert.pem in az client path, however no luck either. So we decided to disable the verification

Should a raise a new bug somewhere

**Comment by yugangw-msft** [comment]
You might have known it already, but here is the [doc](https://github.com/Azure/azure-cli/blob/dev/doc/use_cli_effectively.md#working-behind-a-proxy) for how to work with a proxy

**Comment by savkris** [comment]
thanks @yugangw-msft , 

yes we have tried this as well, but only setting AZURE_CLI_DISABLE_CONNECTION_VERIFICATION would work for us.

**Comment by panchagnula** [comment]
> Okay, so 2 things
> 
> 1. @savkris, your proxy server uses a certificate not fully signed, so you have to by-pass it
> 2. @panchagnula, please forward to Kudu team for why the endpoint `cloudstoragemanager.scm.asemelshared.extnp.nationi` decides to close the connections prematurely.

for Kudu related questions please file an issue here https://github.com/projectkudu/kudu or feel free to raise a support ticket.
